save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/utils/Warning.kt

```diff
@@ -36,35 +41,59 @@ data class Warning(
 @Suppress(
     "TooGenericExceptionCaught",
     "SwallowedException",
-    "ThrowsCount")
+    "ThrowsCount",
+    "TOO_LONG_FUNCTION",
+    "ReturnCount",
+    "ComplexMethod",
+    "NestedBlockDepth",
+    "AVOID_NULL_CHECKS",
+)
 internal fun String.extractWarning(warningRegex: Regex,
                                    fileName: String,
                                    line: Int?,
                                    columnGroupIdx: Int?,
                                    messageGroupIdx: Int,
 ): Warning? {
-    val groups = warningRegex.find(this)?.groups ?: return null
-
-    val column = columnGroupIdx?.let {
-        try {
-            groups[columnGroupIdx]!!.value.toInt()
-        } catch (e: Exception) {
-            throw ResourceFormatException("Could not extract column number from line [$this], cause: ${e.message}")
-        }
+    val groups = warningRegex.find(this)?.groups
+    val defaultLineGroups = if (groups == null) {
+        defaultLinePattern.find(this)?.groups
```

Comment:
Why are you using a hardcoded fixed pattern for default line mode? I see it like this: if `warningRegex` allows empty capture group for line, then we can notice default line mode by checking number of captured groups:
`"// ;warn:?(\\d*):(\\d*): (.+)"`
`// ;warn:12:1: text` - 3 captured groups
`// ;warn:1: text` - still matches the regex, but group 2 is empty -> handle like default line mode

Reply:
Yes. Fixed